internal/subimporter/importer.go

```diff
@@ -707,23 +707,34 @@ func (s *Session) mapCSVHeaders(csvHdrs []string, knownHdrs map[string]bool) map
 // Credit: https://stackoverflow.com/a/24563853
 func countLines(r io.Reader) (int, error) {
 	var (
-		buf     = make([]byte, 32*1024)
-		count   = 0
-		lineSep = []byte{'\n'}
+		buf       = make([]byte, 32*1024)
+		count     = 0
+		lineSep   = byte('\n')
+		lastByte  byte
+		bytesRead int
 	)
 
 	for {
 		c, err := r.Read(buf)
-		count += bytes.Count(buf[:c], lineSep)
-
-		switch {
-		case err == io.EOF:
-			return count, nil
+		if c > 0 {
+			count += bytes.Count(buf[:c], []byte{lineSep})
+			lastByte = buf[c-1]
+			bytesRead += c
```

Comment:
Since `bytesRead` is only checked once (`if bytesRead > 0`)  as a boolean, it's not necessary to count the bytes here. For simplicity, won't simply replacing `if bytesRead > 0 ..` with `lastByte != 0` work?